### ISSUE TITLE
Using SendOption makes networkd crash

### ISSUE THREAD

**Issue by Toniob** [systemd/systemd#14283]
**systemd version the issue has been seen with**
> 244

**Used distribution**
 > Debian sid (unstable)

**Unexpected behaviour you saw**
 > Using SendOption in DHCPv4 section makes networkd crash with segfault

**Steps to reproduce the problem**
Just add a SendOption in DHCPv4 section. Restart systemd-networkd and the process will crash.
For exemple, add `SendOption=12:string:test` in your .network file
And the results in the logs
```
[ 8704.297037] systemd-network[8810]: segfault at 700000001 ip 0000000700000001 sp 00007ffd1b73b9f8 error 14 in systemd-networkd[556067226000+36000]
[ 8704.297046] Code: Bad RIP value.
```

I tried with something else as a type, for exemple ipv4address or uint32, same thing, it crashed.

The result when I launch it with a strace
```
openat(AT_FDCWD, "/etc/systemd/network/eth0.network", O_RDONLY|O_CLOEXEC) = 15
fstat(15, {st_mode=S_IFREG|0644, st_size=356, ...}) = 0
fstat(15, {st_mode=S_IFREG|0644, st_size=356, ...}) = 0
read(15, "[Match]\nName=eth0\n\n[Link]\n#RequiredForOnline=yes\n#Unmanaged=yes\n\n[Network]\nDescription=VDSL Orange\n#IPForward=yes\n#DHCP=no\nDHCP=ipv4\n\n#[Route]\n#Table=100\n\n[DHCPv4]\nSendOption=45:string:test\nUseDNS=no\nUseHostname=no\nUseDomains=no\nClientIdentifier=mac\nVendorClassIdentifier=sagem\nUserClass=FSVDSL_livebox.Internet.softathome.Livebox3\n#CriticalConnection=yes\n", 4096) = 356
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
ioctl(15, TCGETS, 0x7ffed0a7f2a0)       = -1 ENOTTY (Inappropriate ioctl for device)
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x500000001} ---
+++ killed by SIGSEGV +++
```
And the backtrace in gdb
```
#0  0x0000000500000001 in ?? ()
#1  0x00005555555c5787 in ?? ()
#2  0x00005555555c660b in internal_hashmap_remove ()
#3  0x000055555568ae88 in ?? ()
#4  0x0000555555642e18 in ?? ()
#5  0x00005555556434c6 in config_parse ()
#6  0x0000555555643950 in ?? ()
#7  0x0000555555643a8e in config_parse_many ()
#8  0x00005555556ad244 in ?? ()
#9  0x00005555556ae5f8 in ?? ()
#10 0x000055555558c690 in ?? ()
#11 0x00007ffff7e1cbbb in __libc_start_main (main=0x55555558b700, argc=1, argv=0x7fffffffec68, init=<optimized out>, 
    fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffec58) at ../csu/libc-start.c:308
#12 0x000055555558d81a in ?? ()
```

If the option is not correct, the process won't crash. For exemple, with `SendOption=12:` only, I'll get this in the logs
> /etc/systemd/network/eth0.network:12: Invalid DHCP option, ignoring assignment: 12:

Or if the Option number is wrong `SendOption=345:string:test`

> /etc/systemd/network/eth0.network:12: Invalid DHCP option, ignoring assignment: 345:string:test

**Comment by yuwata** [comment]
Thank you for the report. Fix is waiting in #14284. If possible, please test the PR.

**Comment by Toniob** [comment]
Hi,
Thanks for your PR. Unfortunately I won't be able to test it. But I did open a Debian bug : https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=946475 . I'll test it as soon as it is packaged.
